sources/osgShadow/ShadowTexture.js

```diff
@@ -117,19 +117,16 @@ ShadowTexture.prototype = MACROUTILS.objectLibraryClass( MACROUTILS.objectInheri
 
         // update Uniforms
         var uniformMap = this.getOrCreateUniforms( texUnit );
-        uniformMap.ViewMatrix.set( this._viewMatrix );
-        uniformMap.ProjectionMatrix.set( this._projectionMatrix );
-        uniformMap.DepthRange.set( this._depthRange );
-        uniformMap.MapSize.set( this._mapSize );
+        uniformMap.ViewMatrix.setArray( this._viewMatrix );
+        uniformMap.ProjectionMatrix.setArray( this._projectionMatrix );
+        uniformMap.DepthRange.setArray( this._depthRange );
+        uniformMap.MapSize.setArray( this._mapSize );
 
-        this.setDirty( false );
+        this._dirty = false;
```

Comment:
It's useless because the texture will be applied before and the dirty will be already false